### ISSUE TITLE
After an application is rejected, the status updates to "work in progress"

### ISSUE THREAD

**Issue by stellarmagnet** [AutarkLabs/planning-suite#727]
Current behavior:
1) submit application --> status becomes "pending application review"
2) reject application -- > what happens is the status becomes "work in progress"

Expected:
The expected behavior is that if an application is rejected, the status should become "accepting applicants" (if there are no other applications pending review), or "pending application review" if there are applications still pending review. As soon as one of the applicants gets *accepted* is when it will update to work in progress.

This needs to be reflected in both the Issues list and Issue detail

General rules:
* Accepting applicants - Displayed when the bounty is funded and there are NO pending applications, NO work in progress, and NO work ready for review
* Pending application review - Displayed when 1 or more applications are pending review AND when there is NO work in progress and NO work ready for review.
* Work in progress - Displayed when 1 or more applicants have been accepted, and their work has not been submitted yet. 
* Work ready for review - Displayed when 1 or more submissions are pending review.
  * if the work gets rejected, and there were no more applications that haven't been reviewed, it will become "accepting applicants"
  * if the work gets rejected, and there are applicants that haven't been reviewed yet, it will reset to "pending application review"
  * basically, one of the two above is possible, until a new applicant is approved and that is when it becomes "work in progress" again.
